caravel/storage/helpers.py

```diff
@@ -42,9 +42,12 @@ def fetch_shard(shard=""):
     query = entities.Listing.all(keys_only=True).order("-posting_time")
     if shard:
         query = query.filter("keywords =", shard)
+    if email:
```

Comment:
Because Datastore is dumb, this may cause problems. Each "index" is really a large sorted list, so adding a new equality filter here requires resorting the data by email, meaning that this doubles the number of index writes per request.

While it's a little more obtuse, could you rewrite this so that the seller is added to the list of keywords for each entity? (This'd be at https://github.com/uchicago-sg/caravel/blob/manage-listings/caravel/storage/entities.py#L52)